### PR TITLE
add individual video page

### PR DESCRIPTION
**Pull request by iRohitGaur**
[Files to check](https://github.com/iRohitGaur/starflix/pull/6/files?file-filters%5B%5D=.js&file-filters%5B%5D=.jsx&show-viewed-files=true)

[Preview](https://deploy-preview-6--starflix.netlify.app/)

### CODE REVIEW

File: src/context/video-context/video-context.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ function VideoProvider({ children }) {
     filters: [],
   };
 
-  const [state, dispatch] = useReducer((state, action) => {
+  const [videoState, dispatch] = useReducer((state, action) => {
     switch (action.type) {
```
Comment (manojsarna) [inline src/context/video-context/video-context.js]: Suggestion : Be more specific with the naming conventions. Here you have used dispatch.

File: src/pages/video/Video.jsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  const [videoPlayed, setVideoPlayed] = useState(false);
+  const [video, setVideo] = useState(initialVideoState);
+
+  useEffect(() => {
+    videoState.videoData.length !== 0 &&
+      setVideo(videoState.videoData.filter((v) => v._id === videoId)[0]);
+  }, [videoId, videoState, videoState.videoData]);
+
+  const { width: windowWidth } = useWindowDimensions();
+
```
Comment (manojsarna) [inline src/pages/video/Video.jsx:50]: Appreciation : This is nice , you are changing cards on the basis of width. Great Work :)

File: src/pages/video/Video.jsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  const videos = videoState.videoData
+    .filter((v) => v.category === video.category && v._id !== video._id)
+    .slice(0, 4);
+
```
Comment (manojsarna) [inline src/pages/video/Video.jsx]: Appreciation : Showing related videos according to the category.  👍

File: src/pages/video/Video.jsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  const isInLikedVideos =
+    likedVideos.findIndex((v) => v._id === video._id) !== -1;
+
+  const handleAddToPlaylist = (e) => {
+    e.stopPropagation();
+    if (user) {
+      setVideoToAddToPlaylist(video);
+      if (playlists.length === 0) {
+        setCreateNewPlaylistModal(true);
+      } else {
```
Comment (manojsarna) [inline src/pages/video/Video.jsx:105]: Appreciation : Using e.stopPropagation() for event bubling 👍

**Review by itsmebipul00 — APPROVED** [review]
Everything is nicely done. Naming conventions are also awesome. Easy to understand

**Review by manojsarna — APPROVED** [review]
Overall the code the great. Well Written

Looks Great to update  👍

UI Review :

1. Add Scroll To Top - On the video page when I click on the related video it doesnt scroll to the Video Player

image


2. Add to Watch Later 

The button placement for Add to Watch Later <-> Remove to Watach Later it changes!! do check that 

image
image

**Comment by iRohitGaur** [comment]
> Overall the code the great. Well Written
> 
> Looks Great to update 👍
> 
> UI Review :
> 
> 1. Add Scroll To Top - On the video page when I click on the related video it doesnt scroll to the Video Player
> 
> image
> 
> 2. Add to Watch Later
> 
> The button placement for Add to Watch Later <-> Remove to Watach Later it changes!! do check that
> 
> image image

Thanks for the UI suggestions. I will fix the scroll to top.

**Pull request merged**